addOns/network/src/main/java/org/zaproxy/addon/network/internal/client/apachev5/HttpSenderApache.java

```diff
@@ -254,6 +255,9 @@ private void refreshConnectionManager() {
                         .build();
         connectionManager.setDefaultConnectionConfig(connConfig);
 
+        connectionManager.setDefaultSocketConfig(
+                SocketConfig.custom().setSoTimeout(timeout).build());
```

Comment:
For my understanding, what operations exactly does this change affect?

Reply:
The read of a new connection.